### PR TITLE
Meta: Include information about storage in options

### PR DESCRIPTION
**Pull request by fregante**
Screenshot
Screenshot 1

## Related

- https://github.com/fregante/webext-options-sync/issues/27
- https://github.com/refined-github/refined-github/issues/4823

### CODE REVIEW

File: source/options.html [hunk 1]
```diff
@@ -1,6 +1,7 @@
 <!doctype html>
 <meta charset="utf-8">
 <meta name="viewport" content="width=device-width, initial-scale=1, user-scalable=no">
+<base target="_blank">
```
Comment (fregante) [inline source/options.html:4]: Unrelated, but this is now standard in my extensions

File: source/options.html [hunk 2]
```diff
@@ -41,6 +42,8 @@
 		<summary><strong>💅 Custom CSS</strong></summary>
 		<p>Like a userstyle, useful to undo unwanted style changes</p>
 		<p><textarea name="customCSS" rows="2" spellcheck="false"></textarea></p>
+		<p>CSS is limited to 100KB, after which the options <a href="https://github.com/fregante/webext-options-sync/issues/27">will stop being saved</a>.</p>
+		<p>Options storage: <output class="storage-sync">unknown</output></p>
```
Comment (fregante) [inline source/options.html:46]: This is automatically updated on save, but if the user pastes 101KB nothing happens.

I added a link to the issue to help people find it and fix it

File: source/options.tsx [hunk 3]
```diff
@@ -74,6 +75,20 @@ function expandTokenSection(): void {
...
+	const storage = browser.storage[area];
+	const used = await storage.getBytesInUse();
+	// @ts-expect-error Type issue, we should use @types/webextension-polyfill
+	const available = storage.QUOTA_BYTES - used;
+	for (const output of select.all(`.storage-${area}`)) {
+		output.textContent = available < 1000
+			? 'FULL!'
+			: (available < 100_000
+				? `Only ${prettyBytes(available)} available`
+				: `${prettyBytes(used)} used`);
```
Comment (fregante) [inline source/options.tsx:88]: Logic: I don't need to know I have 10MB available, it captures too much attention.

I tried with a `progress` element but most of the times it looks so empty it's unclear.

File: source/options.tsx [hunk 4]
```diff
@@ -260,6 +279,11 @@ function addEventListeners(): void {
 		location.reload();
 	});
 
+	// Update storage usage info
+	browser.storage.onChanged.addListener((_, areaName) => {
+		void updateStorageUsage(areaName as 'sync' | 'local');
```
Comment (fregante) [inline source/options.tsx:284]: They are automatically updated

File: source/options.tsx [hunk 5]
```diff
@@ -281,14 +305,6 @@ function addEventListeners(): void {
 
 	// Add token validation
 	select('[name="personalToken"]')!.addEventListener('input', validateToken);
-
-	// Ensure all links open in a new tab #3181
```
Comment (yakov116) [inline source/options.tsx:285]: Why?

Comment (fregante) [reply source/options.tsx:285]: Moved to `base` element

File: source/options.tsx [hunk 6]
```diff
@@ -297,11 +313,6 @@ async function init(): Promise<void> {
 
 	// TODO: Storage cleanup #6421, Drop in June 2023
 	void browser.storage.local.remove('featuresAlreadySeen');
-
-	// Safari’s storage is inexplicably limited #4823
-	if (isSafari()) {
-		void cache.clear();
-	}
```
Comment (fregante) [inline source/options.tsx:304]: Hopefully they fixed it by now. If not, it will be more visible 🥹

**Pull request merged**